tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -1945,34 +1945,36 @@ public function userHasUploadedFile(
 	}
 
 	/**
-	 * @When /^user "([^"]*)" shares a space "([^"]*)" to user "([^"]*)" with role "([^"]*)"$/
-	 * @When /^user "([^"]*)" shares a space "([^"]*)" to group "([^"]*)" with role "([^"]*)"$/
-	 * @When /^user "([^"]*)" updates the space "([^"]*)" for user "([^"]*)" changing the role to "([^"]*)"$/
+	 * @When /^user "([^"]*)" shares a space "([^"]*)" with settings:$/
+	 * @When /^user "([^"]*)" updates the space "([^"]*)" with settings:$/
 	 *
 	 * @param  string $user
 	 * @param  string $spaceName
-	 * @param  string $recipient
-	 * @param  string $role
+	 * @param  TableNode $table
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 */
 	public function sendShareSpaceRequest(
 		string $user,
 		string $spaceName,
-		string $recipient,
-		string $role
+		TableNode $table
 	): void {
 		$space = $this->getSpaceByName($user, $spaceName);
 		$availableRoleToAssignToShareSpace = ['manager', 'editor', 'viewer'];
-		if (!\in_array(\strtolower($role), $availableRoleToAssignToShareSpace)) {
-			throw new Error("The Selected " . $role . " Cannot be Found");
+		$rows = $table->getRowsHash();
+		if (!\in_array(\strtolower($rows['role']), $availableRoleToAssignToShareSpace)) {
+			throw new Error("The Selected " . $rows['role'] . " Cannot be Found");
 		}
+		$rows["shareType"] = \array_key_exists("shareType", $rows) ? $rows["shareType"] : 7;
+		$rows["expireDate"] = \array_key_exists("expireDate", $rows) ? $rows["expireDate"] : null;
+
 		$body = [
-			"space_ref" => $space['id'],
-			"shareType" => 7,
-			"shareWith" => $recipient,
-			"role" => $role // role overrides the permissions parameter
+			"space_ref" => $space["id"],
+			"shareType" => $rows["shareType"],
+			"shareWith" => $rows["shareWith"],
+			"role" => $rows["role"],
+		  "expireDate" => $rows["expireDate"]
```

Comment:
wrong indent

Reply:
lint drives me crazy

Reply:
it looks good in my IDE and running `make test-php-style-fix` didn't change indent

Reply:
done